web-api/src/storage/elastic/filter.rs

```diff
@@ -117,27 +168,106 @@ mod tests {
 
     use super::*;
 
+    const DATE: &str = "1234-05-06T07:08:09Z";
+
+    #[test]
+    fn test_merge_range() {
+        let mut filter = Vec::new();
+        merge_range(&mut filter);
+        assert!(filter.is_empty());
+
+        let mut filter = vec![json!({})];
+        let expected = filter.clone();
+        merge_range(&mut filter);
+        assert_eq!(filter, expected);
+
+        let mut filter = vec![
+            json!({}),
+            json!({}),
+            json!({ RANGE: { "a": { "gt": "b" } } }),
+            json!({}),
+        ];
+        let expected = filter.clone();
+        merge_range(&mut filter);
+        assert_eq!(filter, expected);
+
+        let mut filter = vec![
+            json!({ RANGE: { "a": { "gt": "b" } } }),
+            json!({ RANGE: { "a": { "lt": "c" } } }),
+            json!({ RANGE: { "a": { "gt": "d" } } }),
+        ];
+        merge_range(&mut filter);
+        assert_eq!(
+            filter,
+            vec![json!({ RANGE: { "a": { "gt": "d", "lt": "c" } } })],
+        );
+
+        let mut filter = vec![
+            json!({}),
+            json!({ RANGE: { "a": { "gt": "b" } } }),
+            json!({ RANGE: { "a": { "lt": "c" } } }),
+            json!({}),
+            json!({}),
+            json!({ RANGE: { "a": { "gt": "d" } } }),
+            json!({}),
+        ];
+        merge_range(&mut filter);
+        assert_eq!(
+            filter,
+            vec![
+                json!({}),
+                json!({ RANGE: { "a": { "gt": "d", "lt": "c" } } }),
+                json!({}),
+                json!({}),
+                json!({}),
+            ],
+        );
```

Comment:
:wrench:  This tests are not correct.

We can not assume if someone writes:

```
{ "$and": [
  { "a": { "$gt":  "b" } },
  { "a": { "$lt":  "c" } },
  { "a": { "$gt":  "d" } },
] }
```

That they mean range `(d;c)`.

It would imply that it's range `(x;c)` where x is the larger date time between `b` and `d` (be aware that this isn't the same as the higher ordered string due to timezones).

Reply:
good catch, i'll update the range merging and also the tests.